src/amazonica/core.clj

```diff
@@ -1030,10 +1030,9 @@
   (->> (.getDeclaredMethods client)
        (remove (fn [method]
                  (let [mods (.getModifiers method)]
-                   (or (.isSynthetic method)
-                       (Modifier/isPrivate mods)
-                       (Modifier/isProtected mods)
-                       (Modifier/isStatic mods)))))
+                   (or (not (Modifier/isPublic mods))
+                       (Modifier/isStatic mods)
+                       (.isSynthetic method)))))
```

Comment:
FWIW: AFAICT `javac` never generates synthetic methods with public visibility, but that's an implementation detail, not a JLS guarantee, so better safe than sorry :)